crypto/fipsmodule/digest/digest.c

```diff
@@ -269,9 +280,16 @@ int EVP_Digest(const void *data, size_t count, uint8_t *out_md,
 
   EVP_MD_CTX_init(&ctx);
   ret = EVP_DigestInit_ex(&ctx, type, impl) &&
-        EVP_DigestUpdate(&ctx, data, count) &&
-        EVP_DigestFinal_ex(&ctx, out_md, out_size);
-  EVP_MD_CTX_cleanup(&ctx);
+        EVP_DigestUpdate(&ctx, data, count);
+  if (EVP_MD_flags(type) & EVP_MD_FLAG_XOF) {
+    if (out_size == NULL) {
+      OPENSSL_PUT_ERROR(DIGEST, ERR_R_PASSED_NULL_PARAMETER);
+      return 0;
+    }
```

Comment:
Can this be moved up to line 280?

Reply:
Sure thing. I've refactored this and added a test.